core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1204,4 +1204,49 @@ public void testExpireWithDeleteFiles() {
             .build(),
         deletedFiles);
   }
+
```

Comment:
Need to add more tests around min snapshots to keep and expire older than for different branches.